src/server/nbs/nbs_frame_codecs.ts

```diff
@@ -0,0 +1,49 @@
+import * as Long from 'long'
+import { NUClearNetPacket } from 'nuclearnet.js'
+
+export const NBS_HEADER = Buffer.from([0xE2, 0x98, 0xA2]) // NUClear radiation symbol.
+
+export type NbsFrame = {
+  // Omitted redundant header information.
+  // header: Buffer,
+  // size: number,
+  timestampInMicroseconds: number,
+  hash: Buffer,
+  payload: Buffer,
+}
+
+// NBS frame format:
+// 3 Bytes - NUClear radiation symbol header, useful for synchronisation when attaching to an existing stream.
+// 4 Bytes - The remaining packet length i.e. 16 bytes + N payload bytes
+// 8 Bytes - 64bit timestamp in microseconds. Note: this is not necessarily a unix timestamp.
+// 8 Bytes - 64bit bit hash of the message type.
+// N bytes - The binary packet payload.
+
+export function encodeFrame(frame: NbsFrame): Buffer {
+  const size = 16 + frame.payload.byteLength
+  const buffer = new Buffer(7 + size)
+  NBS_HEADER.copy(buffer, 0, 0, 3)
```

Comment:
Nodejs has a function Buffer.concat([buffer1,buffer2...]) function. Having each component as it's own buffer and concatenating at the end might remove some of these magic numbers.